spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/FiltersOptimizer.scala

```diff
@@ -0,0 +1,106 @@
+package org.apache.spark.sql.cassandra
+
+import org.apache.spark.sql.sources._
+
+
+/**
+  * Optimizer will try to transform pushdown filter into `sum of products`.
+  * So that the filter like
+  * '(field1 < 3 OR field1 > 7) AND (field2 = 'val1' OR field2 = 'val2')'
+  * will become equivalent
+  * 'field1 < 3 AND field2 = "val1" OR field1 < 3 AND field2 = "val2" OR
+  * field1 > 7 AND field2 = "val1" OR field1 > 7 AND field2 = "val2"'
+  *
+  * @param filters Array of logical statements [[org.apache.spark.sql.sources.Filter]]
+  * that forms `where`-clause with `AND` operator, for example:
+  * val Array(f1, f2, ... fn) = ... // such that `where f1 AND f2 AND ... AND fn`
+  *
+  */
+class FiltersOptimizer(filters: Array[Filter]) {
```

Comment:
I'm a little confused why there is a separate class here, do we ever use this without calling .build() immediately after?